test/__mocks__/@google-cloud/pubsub.ts

```diff
@@ -1,12 +1,24 @@
+import { SubscriptionOptions } from '@google-cloud/pubsub';
+
 type mockedMethods = 'createSubscription' | 'createTopic' | 'topic' | 'subscription';
 type ClientMock = Record<mockedMethods, jest.Mock>;
 
 type SubcriptionMock = Record<'on', jest.Mock | ((...args: any) => any)>;
 
 const initialModule: object = jest.genMockFromModule('@google-cloud/pubsub');
 
-const mockSubscription = (): SubcriptionMock => ({ on: jest.fn() });
-const mockPubSubClient = (sub: SubcriptionMock = mockSubscription()): ClientMock => ({
+const mockSubscription = (): (() => SubcriptionMock) => {
+  let subscriptionOptions: SubscriptionOptions;
+  return () => ({
+    on: jest.fn(),
+    setOptions: (options: SubscriptionOptions) => {
+      subscriptionOptions = options;
+    },
+    getOptions: () => subscriptionOptions,
+  });
+};
+
+const mockPubSubClient = (sub: SubcriptionMock = mockSubscription()()): ClientMock => ({
```

Comment:
I went with a closure to carry an internal state but I'm not sure that's the best / most readable way 😟